internal/validation/mock_trace_validation.go

```diff
@@ -18,55 +18,86 @@ import (
 	"context"
 	"fmt"
 	"reflect"
+	"regexp"
 	"sync"
 	"time"
 
 	"github.com/golang/protobuf/ptypes"
 
 	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
 	"google.golang.org/genproto/googleapis/rpc/errdetails"
-	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+const (
+	maxAnnotationAttributes = 4
```

Comment:
Where do these constants come from? Can we add a link in a comment?

Reply:
They're from https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.cloudtrace.v2, and yep added a comment